src/FSharp.Stats/ML/Unsupervised/KNN.fs

```diff
@@ -0,0 +1,170 @@
+        /// let blues = [| [ 5.0; 6.0 ]; [ 4.0; 5.0 ]; [ 4.0; 6.0 ]; [ 6.0; 6.0 ]; [ 5.0; 4.0 ] |] |> Array.map (fun p -> LabeledPoint<float list, string>.create(p, "blue"))
+        ///
+        /// let labeledPoints = Array.append reds blues
+        /// let prediction = FSharp.Stats.ML.Unsupervised.KNN.Array.predict FSharp.Stats.DistanceMetrics.euclidean labeledPoints 3
+        ///
+        /// let color = prediction [3.0; 3.0] // should be: Some "red"
+        /// let color = prediction [6.0; 6.0] // should be: Some "blue"
+        /// </code> 
+        /// </example>
+        let inline predict (distance : Distance<'a>) (labeledPoints: LabeledPoint<'a, 'l> array) (k : int) (x: 'a) : 'l option =
+            if Array.isEmpty labeledPoints || k <= 0 then
+                None
+            else
+                labeledPoints |> Array.sortInPlaceBy (fun lp -> distance lp.p x)
+
+                let kNearestNeighbors = Array.take k labeledPoints
+
+                let label =
+                    kNearestNeighbors
+                    |> Array.countBy (fun lp -> lp.label)
+                    |> Array.maxBy snd
+                    |> fst
+
+                Some label
+
+    module Seq =
+    
+        /// <summary>
+        /// The [k-nearest neighbors algorithm](https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm) to classify a new data point into the target class, 
+        /// depending on the features of its neighboring data points.
+        /// </summary>
+        /// <param name="distance">the distance function, e.g. `euclidean`</param>
+        /// <param name="labeledPoints">the sequence of classified (or labeled) points, use for the classification</param>
+        /// <param name="k">The number of nearest neighbors from x to look for.</param>
+        /// <param name="x">The point to classify</param>
+        /// <returns>The most common label from the k nearest neighbors for x.</returns>
+        /// <example> 
+        /// <code> 
+        /// let points = seq { [ 2.0; 4.0 ]; [ 1.0; 3.0 ]; [ 2.0; 4.0 ]; [ 3.0; 2.0 ]; [ 2.0; 1.0 ]; [ 5.0; 6.0 ]; [ 4.0; 5.0 ]; [ 4.0; 6.0 ]; [ 6.0; 6.0 ]; [ 5.0; 4.0 ] }        ///
+        /// let labels = seq { "red"; "red"; "red"; "red"; "red"; "blue"; "blue"; "blue"; "blue"; "blue" }
+        /// let prediction = FSharp.Stats.ML.Unsupervised.KNN.Seq.KNN.Seq.predict FSharp.Stats.DistanceMetrics.euclidean points labels 3
+        ///
+        /// let color = prediction [3.0; 3.0] // should be: Some "red"
+        /// let color = prediction [6.0; 6.0] // should be: Some "blue"
+        /// </code> 
+        /// </example>
+        let inline predict<'a, 'l when 'l: equality and 'l: comparison>
+            (distance : Distance<'a>) 
+            (points   : 'a seq)
+            (labels   : 'l seq)
+            (k        : int) 
+            (x        : 'a) 
+            : 'l option =
+
+            if Seq.isEmpty points || Seq.length points <> Seq.length labels || k <= 0 then
```

Comment:
Many thanks for the valuable feedback!
Indeed, I also believe it would be clearer to have a single sequence of tuples, and the array version uses the type 'LabeledPoint' (which is nothing more than a more descriptive tuple).

The reason for this outcome, was the motivation by the existing Python versions, see e.g. [here](https://www.analyticsvidhya.com/blog/2021/04/simple-understanding-and-implementation-of-knn-algorithm/) (Step 2)
which uses the seperate arguments and the need to construct the tuple seq in case you started with seperate data (ofc. the converse argument also holds).

That being said, I am happy to change it as suggested. What do you think?

Reply:
Thanks!
i replaced the seperated params by a tupled sequence.
also i removed the tupe `LabeledPoint` by just tuples, because it seemed as just overhead